terraform/graylog/resource_stream.go

```diff
@@ -139,21 +184,74 @@ func resourceStreamRead(d *schema.ResourceData, m interface{}) error {
 
 func resourceStreamUpdate(d *schema.ResourceData, m interface{}) error {
 	config := m.(*Config)
-	cl, err := client.NewClient(
-		config.Endpoint, config.AuthName, config.AuthPassword)
+	cl, err := client.NewClient(config.Endpoint, config.AuthName, config.AuthPassword)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to create http client: %v", err)
 	}
-	stream, err := newStream(d)
-	if err != nil {
+
+	stream := newStream(d)
+	if err := updateStreamRule(cl, stream); err != nil {
 		return err
 	}
-	if _, err := cl.UpdateStream(stream); err != nil {
+
+	if _, err = cl.UpdateStream(stream); err != nil {
```

Comment:
If only stream rules change and the stream doesn't change, we should not call the update stream API.
We should check the stream changes before call the update stream API.